pkg/controllers/dashboard/helm/repo.go

```diff
@@ -258,23 +263,29 @@ func (r *repoHandler) download(repository *catalog.ClusterRepo, newStatus *catal
 			index, err = git.BuildOrGetIndex(metadata.Namespace, metadata.Name, repoSpec.GitRepo)
 		}
 	} else if repoSpec.GitRepo != "" {
-		commit, err = git.Update(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify, repoSpec.CABundle)
-		if err != nil {
-			retriable = true
-		} else {
-			newStatus.URL = repoSpec.GitRepo
-			newStatus.Branch = repoSpec.GitBranch
-			if newStatus.Commit == commit {
-				newStatus.DownloadTime = downloadTime
-				return setErrorCondition(repository, err, newStatus, interval, repoCondition, r.clusterRepos)
+		// If repo is enabled, then only update the repo.
+		if repoSpec.Enabled == nil || (repoSpec.Enabled != nil && *repoSpec.Enabled) {
+			commit, err = git.Update(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify, repoSpec.CABundle)
+			if err != nil {
+				retriable = true
+			} else {
+				newStatus.URL = repoSpec.GitRepo
+				newStatus.Branch = repoSpec.GitBranch
+				if newStatus.Commit == commit {
+					newStatus.DownloadTime = downloadTime
+					return setErrorCondition(repository, err, newStatus, interval, repoCondition, r.clusterRepos)
+				}
+				index, err = git.BuildOrGetIndex(metadata.Namespace, metadata.Name, repoSpec.GitRepo)
 			}
-			index, err = git.BuildOrGetIndex(metadata.Namespace, metadata.Name, repoSpec.GitRepo)
 		}
 	} else if repoSpec.URL != "" {
-		index, err = helmhttp.DownloadIndex(secret, repoSpec.URL, repoSpec.CABundle, repoSpec.InsecureSkipTLSverify, repoSpec.DisableSameOriginCheck)
-		retriable = true
-		newStatus.URL = repoSpec.URL
-		newStatus.Branch = ""
+		// If repo is enabled, then only update the repo.
+		if repoSpec.Enabled == nil || (repoSpec.Enabled != nil && *repoSpec.Enabled) {
```

Comment:
Why is this condition here instead of in the start of the handler, like in the ensure handler?

Reply:
But is there a reason to create a configMap for a disabled clusterrepo?